bokehjs/src/lib/models/transforms/jitter.ts

```diff
@@ -35,10 +38,22 @@ export class Jitter extends RangeTransform {
   }
 
   override v_compute(xs0: Arrayable<number | Factor>): Arrayable<number> {
-    if (this.previous_values != null && this.previous_values.length == xs0.length)
-      return this.previous_values
-    this.previous_values = super.v_compute(xs0)
-    return this.previous_values
+    let xs: Arrayable<number>
+    if (this.range instanceof FactorRange)
+      xs = this.range.v_synthetic(xs0)
+    else if (isArrayableOf(xs0, isNumber))
+      xs = xs0
+    else
+      unreachable()
+
+    if (this.previous_offsets?.length == xs.length) {
+      const cache_offsets = this.previous_offsets
+      return map(xs, (xs, i) => cache_offsets[i] + xs)
+    }
+
+    const computed_values = map(xs, (xs) => this._compute(xs))
+    this.previous_offsets = map(xs, (xs, i) => computed_values[i] - xs)
```

Comment:
This subtraction is a little awkward to me. I think it would be cleaner if we change `_compute` to compute and return *only the offsets*, instead of the final values. The the logic can be more like:
```
 if (this.previous_offsets?.length != xs.length) {
    this.previous_offsets = this._compute(xs))
}

const {offsets} = this
return map(xs, (xs, i) => xs + offsets[i])
```